### PR TITLE
chore(ci): bun v1

### PR DESCRIPTION
**Pull request by watany-dev**
### Author should do the followings, if applicable

- [ ] Add tests
- [ ] Run tests
- [ ] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by watany-dev** [comment]
After upgrading the version of Bun, the tests related to JSX started to fail. As a temporary fix, I'm importing src/jsx, but I would like to know if there’s a better approach.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: oven-sh/setup-bun@v1
         with:
-          bun-version: '0.6.9'
+          bun-version: '1.x.x'
```
Comment (yusukebe) [inline .github/workflows/ci.yml:58]: We need to fix the version as Bun's API might change frequently.

File: runtime_tests/bun/index.test.tsx [hunk 2]
```diff
@@ -3,6 +3,8 @@ import { serveStatic } from '../../src/adapter/bun'
 import { Context } from '../../src/context'
 import { env, getRuntimeKey } from '../../src/helper/adapter'
 import { Hono } from '../../src/index'
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { jsx } from '../../src/jsx'
```
Comment (yusukebe) [inline runtime_tests/bun/index.test.tsx:7]: It's not a big problem. If a user uses JSX on Bun, they can set it up correctly with `tsconfig.json`.

**Review by yusukebe — COMMENTED** [review]
Hi @watany-dev!

Thank you for the PR. I've left some comments.

File: runtime_tests/bun/index.test.tsx (hunk 2 above)
Comment (watany-dev) [reply runtime_tests/bun/index.test.tsx:7]: I understand that this approach is acceptable, but if there is a need to modify the tsconfig.json, I am more than happy to make the necessary adjustments.

Comment (yusukebe) [reply runtime_tests/bun/index.test.tsx:7]: We don't need to modify the `tsconfig.json` for this repository. Users just need to set up the `tsconfig.json` as shown below, which is already written in the starter for Bun.

```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx",
  }
}
```

**Review by yusukebe — COMMENTED** [review]
LGTM!

**Comment by yusukebe** [comment]
@watany-dev 

Thanks! Let's go with this.

**Pull request merged**